soc/infineon/cat1a/Kconfig.soc

```diff
@@ -0,0 +1,119 @@
+	select CPU_HAS_ARM_MPU  
+	select DYNAMIC_INTERRUPTS
+	select CPU_HAS_FPU
+	select SOC_FAMILY_INFINEON_CAT1
+
+config SOC_FAMILY_PSOC6_LEGACY
+	bool
+	select ARM
+	select HAS_CYPRESS_DRIVERS
+	select CPU_CORTEX_M_HAS_SYSTICK
+	select CPU_HAS_ARM_MPU
+
+config SOC_FAMILY_PSOC6_LEGACY_M4
+	bool
+	select CPU_CORTEX_M4
+	select CPU_CORTEX_M_HAS_DWT
+	select CPU_HAS_FPU
+
+config SOC_FAMILY_PSOC6_LEGACY_M0
+	bool
+	select CPU_CORTEX_M0PLUS
+	select CPU_CORTEX_M_HAS_VTOR
+
+config SOC_PSOC6_M0_ENABLES_M4
+	bool "Dual-core support [activate Cortex-M4]"
+	depends on SOC_FAMILY_PSOC6_LEGACY_M0
+	help
+	  Cortex-M0 CPU should boot Cortex-M4
+
+# Cypress PSoC™ 6 MCU lines
+config SOC_SERIES_PSOC_60
+	bool "Infineon PSoC™ 60 series MCU (Value Line)"
+	select SOC_FAMILY_PSOC6 if !SOC_FAMILY_PSOC6_LEGACY
+	select SOC_FAMILY_INFINEON_CAT1A
+	help
+	  Enable support for Infineon PSoC™ 60 MCU series
+
+config SOC_SERIES_PSOC_61
+	bool "Infineon PSoC™ 61 series MCU (Programmable Line)"
+	select SOC_FAMILY_PSOC6 if !SOC_FAMILY_PSOC6_LEGACY
+	select SOC_FAMILY_INFINEON_CAT1A
+	help
+	  Enable support for Infineon PSoC™ 61 MCU series
+
+config SOC_SERIES_PSOC_62
+	bool "Infineon PSoC™ 62 series MCU (Performance Line)"
+	select SOC_FAMILY_PSOC6 if !SOC_FAMILY_PSOC6_LEGACY
+	select SOC_FAMILY_INFINEON_CAT1A
+	help
+	  Enable support for Infineon PSoC™ 62 MCU series
+
+config SOC_SERIES_PSOC_63
+	bool "Infineon PSoC™ 63 series MCU (Connectivity Line)"
+	select SOC_FAMILY_PSOC6 if !SOC_FAMILY_PSOC6_LEGACY
+	select SOC_FAMILY_INFINEON_CAT1A
+	help
+	  Enable support for Infineon PSoC™ 63 MCU series
+
+config SOC_SERIES_PSOC_64
+	bool "Infineon PSoC™ 64 series MCU (Security Line)"
+	select SOC_FAMILY_PSOC6 if !SOC_FAMILY_PSOC6_LEGACY
+	select SOC_FAMILY_INFINEON_CAT1A
+	help
+	  Enable support for Infineon PSoC™ 64 MCU series
+
+config SOC_SERIES
+	default "psoc6_60" if SOC_SERIES_PSOC_60
+	default "psoc6_61" if SOC_SERIES_PSOC_61
+	default "psoc6_62" if SOC_SERIES_PSOC_62
+	default "psoc6_63" if SOC_SERIES_PSOC_63
+	default "psoc6_64" if SOC_SERIES_PSOC_64
+
+if SOC_FAMILY_PSOC6
+	## PSoC™ 6 Cortex M0+ prebuilt images
+	choice
+		prompt "PSoC™ 6 Cortex M0+ prebuilt images"
+		help
+		  Choose the prebuilt application image to be executed on the Cortex-M0+ core of the PSoC™ 6
+		  dual-core MCU. The image is responsible for booting the Cortex-M4 on the device.
+
+	config SOC_PSOC6_CM0P_IMAGE_SLEEP
+		bool "DeepSleep"
+		help
+		  DeepSleep prebuilt application image is executed on the Cortex-M0+ core of the PSoC™ 6 BLE
+		  dual-core MCU.The image is provided as C array ready to be compiled as part of the Cortex-M4
+		  application. The Cortex-M0+ application code is placed to internal flash by the Cortex-M4
+		  linker script.
+	endchoice
+
+	config SOC_PSOC6_CM0P_IMAGE_ROM_SIZE
+	hex
+	default 0x2000 if SOC_PSOC6_CM0P_IMAGE_SLEEP
+
+	config SOC_PSOC6_CM0P_IMAGE_RAM_SIZE
+	hex
+	default 0x2000 if SOC_PSOC6_CM0P_IMAGE_SLEEP
+
+endif # SOC_FAMILY_PSOC6
```

Comment:
indent completely wrong, goes in Kconfig

Reply:
updated. moved to Kconfig (from Kconfig.soc), removed indent.